「ReactでSPAを作る」のプログラムを作成しました

a-kuroki-gs opened the pull request:
フィヨルドブートキャンプの[ReactでSPAを作る](https://bootcamp.fjord.jp/practices/262)の課題である、[メモアプリSPA版](https://bootcamp.fjord.jp/pages/239)を作成しました。

a-kuroki-gs committed:
メモの表示ができるように実装

a-kuroki-gs committed:
新規メモ作成を親コンポーネントで実装

a-kuroki-gs committed:
詳細→編集・削除のボタンを実装

a-kuroki-gs committed:
詳細画面でメモの内容を表示できるようにする

a-kuroki-gs committed:
メモの内容をtextとして保持するように修正

a-kuroki-gs committed:
クリックするとそのメモの編集画面が表示できるようにする

a-kuroki-gs committed:
見た目を整える

a-kuroki-gs committed:
テストデータの消去

a-kuroki-gs committed:
lintを通す

a-kuroki-gs committed:
保存先ストレージをLocalStorageに設定する

a-kuroki-gs committed:
不要なコメントの削除

a-kuroki-gs committed:
eslintの設定の修正

reactのバージョンを指定していないというエラーが出ていたので

a-kuroki-gs committed:
メモを削除した後に、何も選択していない状態にする

削除したテキストが表示されないようにするため

a-kuroki-gs committed:
変数名の変更

a-kuroki-gs committed:
READMEの内容を作成

yuuu commented on README.md:
```diff
@@ -1 +1,33 @@
 # react-memo-app
+
+## 概要
```
せっかくなので開発者向けに、このリポジトリをクローンしてから手元で起動するまでの手順も記載されているとさらに良いですね 👍

a-kuroki-gs replied:
READMEにクローンから起動までの手順を追加しました。

yuuu commented on public/index.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!doctype html>
```
ここは大文字で書くことが多いと感じているのですが、修正した理由を教えていただけますでしょうか？ESLintでしょうか？

a-kuroki-gs replied:
こちらはPrettierの指摘にともなって修正した部分になります。

[補足](https://bootcamp.fjord.jp/products/18546#-3)で詳しく回答いたしました！
(Prettier 3.0から小文字を採用するようになったそうです。)

yuuu replied:
なるほど。これは勉強になりました 👀 補足までいただきありがとうございます 🙏

yuuu commented on src/App.js:
```diff
@@ -1,10 +1,76 @@
+  });
+  const [selectedMemo, setSelectedMemo] = useState(null);
+
+  function saveToLocalStorage(data) {
```
`data` はあらゆる変数名に該当しうる名前なので `memos` のような名前はどうでしょうか？

a-kuroki-gs replied:
ご指摘いただいたように`memos`に修正しました。

yuuu commented on src/App.js:
```diff
@@ -1,10 +1,76 @@
+
+  function handleNewMemo() {
+    const newMemo = {
+      id: memos.length + 1,
```
このidの付け方だと、途中のメモを削除した場合に同じidが付与される可能性がありますね。何かしらの対策をお願い致します 🙏

a-kuroki-gs replied:
元々`memos.length`で管理していたためこの問題が発生しておりました。

新たに`maxId`という変数を導入し、
`maxId`を`memos`のIDの最大値とし、新規に作成するメモのIDを`maxId+1`とすることで対応しました。

a-kuroki-gs committed:
READMEの修正

クローンしてから手元で起動するまでの手順を追加

a-kuroki-gs committed:
変数名の修正

dataという意味のない変数名になっていたため

a-kuroki-gs committed:
idの付け方を修正

現状だと途中のメモの削除後、同じidが付与されてしまう可能性があるため
maxIdをmemosのIDの最大値とし、新規に作成するメモのIDをmaxId+1にする

yuuu commented on README.md:
````diff
@@ -1 +1,54 @@
+```
+
+そして、以下のコマンドを実行すると、メモアプリが起動します。
+(必要なnpmパッケージがあれば、適宜インストールしてください。)
````
ここは「 `npm install` を実行してください」で問題ないかと 👍

a-kuroki-gs replied:
修正しました！

a-kuroki-gs committed:
READMEの修正

npmパッケージをインストールする手順で、「npm install を実行してください」と記載する
